aea/configurations/base.py

```diff
@@ -544,17 +642,20 @@ def __init__(self,
     def default_connection(self) -> str:
         """Get the default connection."""
         assert self._default_connection is not None, "Default connection not set yet."
-        return self._default_connection
+        return str(self._default_connection)
 
     @default_connection.setter
-    def default_connection(self, connection_name: str):
+    def default_connection(self, connection_name: Optional[str]):
```

Comment:
Not sure why this is needed as Optional?

Reply:
the type of `self._default_connection` (after the changes) is `Optional[PublicId]`, but the setter didn't accept `None` values.

(the same holds for the previous code, where the type was simply `str`)